test/check-machines-create

```diff
@@ -2321,6 +2322,182 @@ vnc_password= "{vnc_passwd}"
                                               "expired",
                                               False)
 
+    @contextlib.contextmanager
+    def remove_spice_plugins(self):
+        m = self.machine
+        # avoid uninstalling packages to keep this nondestructive
+        plugins = m.execute("find /usr/lib* -path '*qemu*spice*' | xargs").strip()
+        try:
+            m.execute(f"mkdir {self.vm_tmpdir}/qemu; for f in {plugins}; do mv $f {self.vm_tmpdir}/qemu; done")
+            caps = m.execute("virsh domcapabilities")
+            if m.image.startswith("debian") or m.image.startswith("ubuntu"):
+                # on Debian, spice graphics is built in, spicevmc channels are a plugin
+                self.assertNotIn("spicevmc", caps)
+            else:
+                # in Fedora/RHEL we don't expect any remaining capability
+                self.assertNotIn("spice", caps)
+
+            yield
```

Comment:
What does this `yield` do?

Reply:
Oh, I get it, context manager! Fancy!

Reply:
Right, you can mentally replace it with the code block inside the `with:`